osu.Game/Screens/Play/KeyCounter.cs

```diff
@@ -1,58 +1,56 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
-#nullable disable
-
-using osu.Framework.Allocation;
+using osu.Framework.Bindables;
 using osu.Framework.Graphics;
 using osu.Framework.Graphics.Containers;
-using osu.Framework.Graphics.Sprites;
-using osu.Framework.Graphics.Textures;
-using osu.Game.Graphics;
-using osu.Game.Graphics.Sprites;
-using osuTK;
-using osuTK.Graphics;
 
 namespace osu.Game.Screens.Play
 {
     public abstract partial class KeyCounter : Container
     {
-        private Sprite buttonSprite;
-        private Sprite glowSprite;
-        private Container textLayer;
-        private SpriteText countSpriteText;
+        public readonly InputTrigger Trigger;
 
-        public bool IsCounting { get; set; } = true;
-        private int countPresses;
+        protected Bindable<bool> IsCountingBindable = new BindableBool(true);
 
-        public int CountPresses
+        private readonly Container content;
+
+        protected override Container<Drawable> Content => content;
+
+        protected Bindable<int> PressesCount = new BindableInt
         {
-            get => countPresses;
-            private set
-            {
-                if (countPresses != value)
-                {
-                    countPresses = value;
-                    countSpriteText.Text = value.ToString(@"#,0");
-                }
-            }
+            MinValue = 0
+        };
+
+        public bool IsCounting
+        {
+            get => IsCountingBindable.Value;
+            set => IsCountingBindable.Value = value;
         }
 
-        private bool isLit;
+        public int CountPresses
+        {
+            get => PressesCount.Value;
+            private set => PressesCount.Value = value;
+        }
 
-        public bool IsLit
+        protected KeyCounter(InputTrigger trigger)
         {
-            get => isLit;
-            protected set
+            InternalChildren = new Drawable[]
             {
-                if (isLit != value)
+                content = new Container
                 {
-                    isLit = value;
-                    updateGlowSprite(value);
-                }
-            }
+                    RelativeSizeAxes = Axes.Both
+                },
+                Trigger = trigger,
+            };
+
+            Trigger.Target = this;
+            Name = trigger.Name;
         }
 
+        protected Bindable<bool> IsLit = new BindableBool();
```

Comment:
I'd depart from the "lighting:" vernacular here. I'm not sure it's going to make sense for all implementations ever. My proposal:

- `IsLit` -> `IsActive`
- `Light()` -> `Activate()`
- `Unlight()` -> `Deactivate()`